quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/storage/database/jpa/QuarkusJpaConnectionProviderFactory.java

```diff
@@ -322,8 +327,15 @@ private void importRealms() {
                 String file = tokenizer.nextToken().trim();
                 RealmRepresentation rep;
                 try {
+                    Path filePath = Paths.get(file);
```

Comment:
Giving this PR another look, one question around this line and the following check for "json":

In case of a symlink, does `filePath` contain the symlinkspath/name/ending or the actual files path and name/extension? 

Maybe wrong here, but when that one contains just the symlink, it would be theoretically possible to do a symlink from "import.json" pointing to "myBadHackingScript.java" right? So just want to make sure we check the ending of the actual file that's linked to here, not the one of the symlink.

i mean with the above that I am not sure we should use Paths.toRealPath instead of Paths.get before checking the file ending, because it may (or may not when I understand it wrong) be that we're just checking the symlink extension instead of the file extension and thus opening attack vectors where there's no need to. 

ref: https://docs.oracle.com/javase/8/docs/api/java/nio/file/Path.html#toRealPath-java.nio.file.LinkOption...-

Reply:
For me, it makes not much difference but if you really think this is needed perhaps you can elaborate more about the attack vectors?

Reply:
I don't have concrete attack vectors in mind, that's why its a general question. 

_if my assumption is true_, it just seemed a bit weird for me that we are checking only that the symlink is ending with `json` without checking that the actual file the symlink links to is a json file. 

Afterwards we try to serialize it, so if there's any bug in the serialization, when you add a specifically crafted sh/exe file or sth, we would be open for it. But as said, I don't have an exact CVE in mind. Perhaps @abstractj knows more? 

I am not hard on this, question is just why not add it as the change looks like it is simple to do and will definitely avoid the scenario i outline above?

Reply:
works for me for now, approved.